domain/src/main/java/no/difi/meldingsutveksling/receipt/Conversation.java

```diff
@@ -1,166 +1,162 @@
 package no.difi.meldingsutveksling.receipt;
 
-import com.fasterxml.jackson.annotation.JsonFormat;
 import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.google.common.base.MoreObjects;
-import com.google.common.collect.Lists;
-import lombok.Data;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
+import no.difi.meldingsutveksling.MessageInformable;
 import no.difi.meldingsutveksling.ServiceIdentifier;
-import no.difi.meldingsutveksling.core.EDUCore;
-import no.difi.meldingsutveksling.domain.sbdh.EduDocument;
+import no.difi.meldingsutveksling.nextmove.AbstractEntity;
 import no.difi.meldingsutveksling.nextmove.ConversationDirection;
-import no.difi.meldingsutveksling.nextmove.ConversationResource;
-import no.difi.meldingsutveksling.noarkexchange.PayloadException;
-import no.difi.meldingsutveksling.noarkexchange.PayloadUtil;
+import org.hibernate.annotations.UpdateTimestamp;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.persistence.*;
-import java.time.LocalDateTime;
-import java.util.List;
-import java.util.stream.Stream;
+import java.time.OffsetDateTime;
+import java.util.HashSet;
+import java.util.Set;
 
-import static no.difi.meldingsutveksling.ServiceIdentifier.DPE_INNSYN;
-import static no.difi.meldingsutveksling.ServiceIdentifier.DPV;
 import static no.difi.meldingsutveksling.receipt.ConversationMarker.markerFrom;
 
+@Getter
+@Setter
 @Entity
-@Data
 @Slf4j
 @Table(name = "conversation",
-    indexes = {@Index(columnList = "conversation_id")})
-public class Conversation {
+        indexes = {@Index(columnList = "conversation_id")})
+@ApiModel(description = "Conversation")
+@NamedEntityGraph(name = "Conversation.messageStatuses", attributeNodes = @NamedAttributeNode("messageStatuses"))
+public class Conversation extends AbstractEntity<Long> implements MessageInformable {
 
     public static final Logger statusLogger = LoggerFactory.getLogger("STATUS");
 
-    @Id
-    @GeneratedValue
-    private Integer convId;
+    @Override
+    @JsonProperty
+    @ApiModelProperty(
+            position = 2,
+            value = "Id",
+            example = "1")
+    public Long getId() {
+        return super.getId();
+    }
+
     @Column(name = "conversation_id")
     private String conversationId;
+    @Column(name = "message_id")
+    private String messageId;
     private String senderIdentifier;
     private String receiverIdentifier;
     private String messageReference;
     private String messageTitle;
-    @JsonFormat(shape = JsonFormat.Shape.STRING)
-    private LocalDateTime lastUpdate;
+    private String serviceCode;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1213)